main.go

```diff
@@ -148,6 +182,27 @@ func main() {
 			return
 		}
 
+		//If proxy is set to kubernetes service(apiserver) add SA token to the request
+		if strings.Contains(cfg.upstream, "kubernetes.default") {
```

Comment:
This seems very hardcoded to me and I don't seem to understand the use case completely. Do I understand it correctly that you are putting an RBAC proxy before the API server? I am unsure we want to support this use case here as the API server itself takes care of authn/authz already.

Reply:
The motivation for this was a feature allowing to secure api-server in the environments where admins cannot modify api-server configurations like OIDC. But I do accept that this is not desired feature and I will remove it